engine/servermaster/service_util.go

```diff
@@ -42,3 +50,9 @@ func newMultiClient(conn *grpc.ClientConn) multiClient {
 		JobManagerClient:      enginepb.NewJobManagerClient(conn),
 	}
 }
+
+func generateNodeID(name string) string {
+	val := rand.Uint32()
+	id := fmt.Sprintf("%s-%08x", name, val)
```

Comment:
What if the name is empty?

Reply:
The master and executor will generate a default name for themselves, so the name will not be empty.